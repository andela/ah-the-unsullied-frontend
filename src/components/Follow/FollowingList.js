import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import '../../assets/styles/HomePage.scss';
import { followingList } from '../../actions/profile/followActions';

export class FollowingList extends Component {
  componentDidMount() {
    const { username } = this.props;
    this.props.followingList(username);
  }

  render() {
    const list = this.props.Following.following;
    return (
      <div>
        <div>
          <span>
            <span className="number">{list.length}</span>
            Following
          </span>
        </div>
      </div>
    );
  }
}

FollowingList.propTypes = {
  Following: PropTypes.object.isRequired,
  followingList: PropTypes.func.isRequired,
  username: PropTypes.string.isRequired
};

export const mapStateToProps = state => ({
  auth: state.auth,
  Following: state.following
});

export default connect(
  mapStateToProps,
  { followingList }
)(FollowingList);
