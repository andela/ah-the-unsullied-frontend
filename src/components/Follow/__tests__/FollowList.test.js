import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16/build';
import { FollowingList, mapStateToProps } from '../FollowingList';

Enzyme.configure({ adapter: new Adapter() });
describe('follow button', () => {
  let props;
  let wrapper;
  let wrapperInstance;
  let state;
  beforeEach(() => {
    props = {
      bio: 'i have abio',
      username: 'nesh',
      image: 'hshhshs',

      auth: {
        isAuthenticated: true,
        user: {
          username: 'nesh'
        }
      },
      Following: {
        following: {}
      },
      list: [],

      followingList: jest.fn(() => {
        Promise.resolve();
      })
    };
    

    state = {
      auth: {},
      Following: {
        following: {}
      }
    };
    wrapper = shallow(<FollowingList {...props} />);
    wrapperInstance = wrapper.instance();
  });

  it('should render the follow list', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('mapStateToProps', () => {
    const propsState = mapStateToProps(state);
    expect(propsState.auth).toEqual(state.auth);
  });
});
