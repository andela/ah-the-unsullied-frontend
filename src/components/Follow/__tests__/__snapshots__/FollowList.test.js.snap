// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`follow button should render the follow list 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FollowingList
    Following={
      Object {
        "following": Object {},
      }
    }
    auth={
      Object {
        "isAuthenticated": true,
        "user": Object {
          "username": "nesh",
        },
      }
    }
    bio="i have abio"
    followingList={
      [MockFunction] {
        "calls": Array [
          Array [
            "nesh",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    image="hshhshs"
    list={Array []}
    username="nesh"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <span>
          <span
            className="number"
          />
          Following
        </span>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span>
          <span
            className="number"
          />
          Following
        </span>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="number"
            />,
            "Following",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "number",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          "Following",
        ],
        "type": "span",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <span>
            <span
              className="number"
            />
            Following
          </span>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span>
            <span
              className="number"
            />
            Following
          </span>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="number"
              />,
              "Following",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "number",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            "Following",
          ],
          "type": "span",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
